lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
```

Comment:
I saw this validation in the code and assume that it could happen. but it seems to me like this shouldn't happen.